packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -129,51 +136,70 @@ contract SuperchainConfig is Initializable, ISemver {
     function initiator() public view returns (address initiator_) {
-        initiator_ = Slot.getAddress(INITIATOR_SLOT);
+        initiator_ = Storage.getAddress(INITIATOR_SLOT);
     }
 
     /// @notice Getter for the vetoer address.
     function vetoer() public view returns (address vetoer_) {
-        vetoer_ = Slot.getAddress(VETOER_SLOT);
+        vetoer_ = Storage.getAddress(VETOER_SLOT);
     }
 
     /// @notice Getter for the guardian address.
     function guardian() public view returns (address guardian_) {
-        guardian_ = Slot.getAddress(GUARDIAN_SLOT);
+        guardian_ = Storage.getAddress(GUARDIAN_SLOT);
     }
 
     /// @notice Getter for the delay time.
     function delay() public view returns (uint256 delay_) {
-        delay_ = Slot.getUint(DELAY_SLOT);
+        delay_ = Storage.getUint(DELAY_SLOT);
     }
 
-    /// @notice Getter for the maxPause time.
+    /// @notice Getter for the maxPause duration.
     function maxPause() public view returns (uint256 maxPause_) {
-        maxPause_ = Slot.getUint(MAX_PAUSE_SLOT);
+        maxPause_ = Storage.getUint(MAX_PAUSE_SLOT);
     }
 
-    /// @notice Getter for the paused address.
+    /// @notice Getter for the current paused status.
     function paused() public view returns (bool paused_) {
-        paused_ = Slot.getUint(PAUSED_TIME_SLOT) > block.timestamp;
+        paused_ = Storage.getUint(PAUSED_TIME_SLOT) > block.timestamp;
+    }
+
+    /// @notice Getter for the paused until time.
+    function pausedUntil() public view returns (uint256 paused_) {
+        paused_ = Storage.getUint(PAUSED_TIME_SLOT);
     }
 
-    /// @notice Pauses withdrawals.
-    function pause(uint256 duration) external {
+    /// @notice Pauses withdrawals by the specified duration.
+    ///         If already paused, the end of the pause period will be delayed by the specified duration.
+    /// @param duration The duration of the pause.
+    /// @param identifier (Optional) A string to identify provenance of the pause transaction.
+    function pause(uint256 duration, string memory identifier) external {
         require(msg.sender == guardian(), "SuperchainConfig: only guardian can pause");
-        require(duration <= Slot.getUint(MAX_PAUSE_SLOT), "SuperchainConfig: duration exceeds maxPause");
-        Slot.setUint(PAUSED_TIME_SLOT, uint256(block.timestamp) + duration);
-        emit Paused();
+        require(duration <= Storage.getUint(MAX_PAUSE_SLOT), "SuperchainConfig: duration exceeds maxPause");
+
+        if (paused() == false) {
+            // Pause is not active, so set the pause end time to the current time plus the duration.
+            Storage.setUint(PAUSED_TIME_SLOT, uint256(block.timestamp) + duration);
+            emit Paused(duration, identifier);
+        } else {
+            // Pause is already active, so extend the current pause end time by the duration.
+            uint256 currentPausedUntil = Storage.getUint(PAUSED_TIME_SLOT);
```

Comment:
naming nit: could remove intermediate variable here by doing read and addition in a single expression but not a big deal

Reply:
In the past you've preferred to avoid nested expressions. I think it depends on the situation. 

Anyway, I looked at the code and I agree it's nicer to avoid these intermediate vars.
Changing that now.